tests/Unit/Models/FamilyModelTest.php

```diff
@@ -6,12 +6,128 @@
 use App\Carer;
 use App\Child;
 use App\Centre;
+use App\Registration;
 use Illuminate\Foundation\Testing\DatabaseMigrations;
 
 class FamilyModelTest extends TestCase
 {
     use DatabaseMigrations;
 
+    /** @test */
+    public function itCreditsWhenAFamilyIsPregnant()
+    {
+        // Make a pregnant family
+        $family = factory(Family::class)->create();
+
+        $pregnancy = factory(Child::class, 'unbornChild')->make();
+        $family->children()->save($pregnancy);
+
+        // There should be a credit reason of 'FamilyIsPregnant'
+        $credit_reasons = $family->getCreditReasons();
+        $this->assertEquals(1, count($credit_reasons));
+        $this->assertEquals('pregnant', $credit_reasons[0]["reason"]);
+    }
+
+    /** @test */
+    public function itCanHaveRegistrations()
+    {
+        // Create Family
+        $family = factory(Family::class)->create();
+
+        // There should be no registrations, and that's fine.
+        $this->assertEquals(0, $family->registrations()->count());
+
+        // Associate with two registrations
+        $registration1 = new Registration();
+        $registration1->family_id = $family->id;
+        $registration1->centre_id = factory(Centre::class)->create()->id;
+        $registration1->eligibility = "other";
+        $registration1->save();
+
+        $registration2 = new Registration();
+        $registration2->family_id = $family->id;
+        $registration2->centre_id = factory(Centre::class)->create()->id;
+        $registration2->eligibility = "other";
+        $registration2->save();
+
+        $registrations = [$registration1, $registration2];
+
+        // Check they're there.
+        $this->assertEquals(2, $family->registrations->count());
+        foreach ($family->registrations as $index => $registration) {
+            $this->assertEquals($registration->id, $registrations[$index]->id);
+        }
+    }
+
+        /** @test */
+    public function itCanHaveCarers()
+    {
+        // Create Family
+        $family = factory(Family::class)->create();
+
+        // There should be no carers, and that's fine.
+        $this->assertEquals(0, $family->carers()->count());
+
+        // Add 3 Carers
+        $carers = factory(Carer::class, 3)->make();
+        $family->carers()->saveMany($carers);
+
+        // Check they're there.
+        $this->assertEquals(3, $family->carers->count());
+        foreach ($family->carers as $index => $carer) {
+            $this->assertEquals($carer->id, $carers[$index]->id);
+        }
+    }
+
+    /** @test */
+    public function itCanHaveChildren()
+    {
+        // Create Family
+        $family = factory(Family::class)->create();
+
+        // There should be no kids, and that's fine.
+        $this->assertEquals(0, $family->children()->count());
+
+        // Add 3 Children
+        $children = factory(Child::class, 3)->make();
+        $family->carers()->saveMany($children);
```

Comment:
Should this be children() rather than carers() ?

Reply:
oops. yup!